drivers/usb/udc/Kconfig.dwc2

```diff
@@ -17,6 +17,13 @@ config UDC_DWC2_DMA
 	help
 	  Enable Buffer DMA if DWC2 USB controller supports Internal DMA.
 
+config UDC_DWC2_HIBERNATION
+	bool "DWC2 USB Hibernation support"
+	default y
+	depends on UDC_DWC2
+	help
+	  Enable Hibernation if DWC2 USB controller supports hibernation.
```

Comment:
What would be a reason to disable it? Please add to the help description.

Reply:
This is generally to allow user to disable hibernation in case:
  * there is some hardware bug in implementation (and there is no quirk implemented yet)
  * there is some bug in USB stack/driver that triggers when hibernation is enabled (and the bug is not yet fixed)

In both cases the intended use case is to workaround issues. This also gives the user the possibility to determine if it is hibernation-related or not before reporting bug.